WalletWasabi/Services/WalletService.cs

```diff
@@ -802,6 +732,97 @@ public async Task<Block> GetOrDownloadBlockAsync(uint256 hash, CancellationToken
 			return block;
 		}
 
+		private async Task<Block> DownloadBlockFromLocalNodeAsync(uint256 hash, CancellationToken cancel)
+		{
+			try
+			{
+				if (LocalBitcoinCoreNode is null || !LocalBitcoinCoreNode.IsConnected && Network != Network.RegTest) // If RegTest then we're already connected do not try again.
+				{
+					DisconnectDisposeNullLocalBitcoinCoreNode();
+					using (var handshakeTimeout = CancellationTokenSource.CreateLinkedTokenSource(cancel))
+					{
+						handshakeTimeout.CancelAfter(TimeSpan.FromSeconds(10));
+						var nodeConnectionParameters = new NodeConnectionParameters()
+						{
+							ConnectCancellation = handshakeTimeout.Token,
+							IsRelay = false,
+							UserAgent = $"/Wasabi:{Constants.ClientVersion}/"
+						};
+
+						// If an onion was added must try to use Tor.
+						// onlyForOnionHosts should connect to it if it's an onion endpoint automatically and non-Tor endpoints through clearnet/localhost
+						if (Synchronizer.WasabiClient.TorClient.IsTorUsed)
+						{
+							nodeConnectionParameters.TemplateBehaviors.Add(new SocksSettingsBehavior(Synchronizer.WasabiClient.TorClient.TorSocks5EndPoint, onlyForOnionHosts: true, networkCredential: null, streamIsolation: false));
+						}
+
+						var localEndPoint = ServiceConfiguration.BitcoinCoreEndPoint;
+						var localNode = await Node.ConnectAsync(Network, localEndPoint, nodeConnectionParameters);
+						try
+						{
+							Logger.LogInfo<WalletService>("TCP Connection succeeded, handshaking...");
+							localNode.VersionHandshake(Constants.LocalNodeRequirements, handshakeTimeout.Token);
+							var peerServices = localNode.PeerVersion.Services;
+
+							//if (!peerServices.HasFlag(NodeServices.Network) && !peerServices.HasFlag(NodeServices.NODE_NETWORK_LIMITED))
+							//{
+							//	throw new InvalidOperationException($"Wasabi cannot use the local node because it does not provide blocks.");
+							//}
```

Comment:
Should this commented code be removed?

Reply:
Was there beforehand, most of this is just copy-pasting

Reply:
It can go I think. @lontivero ?